libs/core/src/lib/title/title.component.ts

```diff
@@ -0,0 +1,28 @@
+import { ChangeDetectionStrategy, Component, ElementRef, Input, OnInit, ViewEncapsulation } from '@angular/core';
+
+@Component({
+    // tslint:disable-next-line
+    selector: 'h1[fd-title], h2[fd-title], h3[fd-title], h4[fd-title], h5[fd-title], h6[fd-title]',
+    exportAs: 'fd-title',
+    template: `<ng-content></ng-content>`,
+    host: {
+        class: 'fd-title',
+        '[class.fd-title--wrap]': 'wrap'
+    },
+    styleUrls: ['./title.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class TitleComponent implements OnInit {
+    /** Whether or not the title should wrap (truncates by default) */
+    @Input()
+    wrap = false;
+
+    constructor(private _elRef: ElementRef) {}
+
+    /** @hidden */
+    ngOnInit(): void {
+        const headerSize = this._elRef.nativeElement.tagName.charAt(1);
+        this._elRef.nativeElement.classList.add('fd-title--h' + headerSize);
```

Comment:
I think in this case the user should pass the value for the header. The way it's now (`Visual` example) the compiled code is:
<img width="637" alt="Screen Shot 2020-10-30 at 12 22 25 PM" src="https://user-images.githubusercontent.com/39598672/97730900-d7aaf580-1aaa-11eb-8d7e-2226514b269e.png">
The html tag matches the `fd-title--hX` class. 
In styles it's different:
<img width="1249" alt="Screen Shot 2020-10-30 at 12 22 37 PM" src="https://user-images.githubusercontent.com/39598672/97730984-f3ae9700-1aaa-11eb-8d93-a718e4686291.png">

The idea is that semantically it would make sense to use `h1`, for example, but the user might want to change the default font-size and have something like:
`<h1 class="fd-title fd-title--h6">Title Level 1</h1>`